packages/flutter_tools/test/src/process.dart

```diff
@@ -0,0 +1,144 @@
+    Encoding? stderrEncoding = systemEncoding,
+  }) {
+    final ProcessResult result = runSync(
+      command,
+      workingDirectory: workingDirectory,
+      environment: environment,
+      includeParentEnvironment: includeParentEnvironment,
+      stdoutEncoding: stdoutEncoding,
+      stderrEncoding: stderrEncoding,
+    );
+    if (result.exitCode != 0) {
+      throw Exception(
+        'child process exited with code ${result.exitCode}\n'
+        'command: ${shellEscapeCommand(command)}\n'
+        'stdout: ================================================================\n'
+        '${result.stdout}\n'
+        'stderr: ================================================================\n'
+        '${result.stderr}\n'
+        'end ====================================================================',
+      );
+    }
+    return result;
+  }
+}
+
+final RegExp _trailingNewlineRegExp = RegExp(r'\n*$');
+
+String _asShellOutput(String raw) {
+  return raw.replaceFirst(_trailingNewlineRegExp, '');
+}
+
+List<int> _bytesAsShellOutput(List<int> raw) {
+  final int end = 1 + raw.lastIndexWhere((int byte) => byte != 0x0a);
+  return end == raw.length ? raw : raw.sublist(0, end);
+}
+
+extension FileExtension on File {
+  /// Reads the file contents as a string with the semantics of `$(cat …)`,
+  /// returning null if the operation fails.
+  ///
+  /// This is commonly the intended semantics when a file was meant to be
+  /// read or written by a shell script.
+  ///
+  /// The file's contents are read using the given [Encoding], and then
+  /// any run of newlines at the end of the string is removed.
+  ///
+  /// See also:
+  /// * [ProcessResultExtension.shellOutput], for the semantics of `$(…)`
+  ///   on an arbitrary command.
+  /// * the Bash manual on command substitution `$(…)`: <https://www.gnu.org/software/bash/manual/bash.html#Command-Substitution>.
+  String? readStringLikeShell({Encoding encoding = utf8}) {
+    try {
+      return _asShellOutput(readAsStringSync(encoding: encoding));
+    } on FileSystemException {
+      return null;
+    }
+  }
+}
+
+extension ProcessResultExtension on ProcessResult {
+  /// The command's output, as shell command substitution `$(…)` would take it.
+  ///
+  /// Among commands following Unix CLI conventions, this is commonly the
+  /// intended semantics for consuming the output.
+  ///
+  /// This is defined as the result of removing from [stdout] any run of
+  /// newlines at the end of the string.  For example, if [stdout] is any of
+  /// 'a\nb', 'a\nb\n', or 'a\nb\n\n\n', then [shellOutput] will be 'a\nb'.
+  ///
+  /// This value has the same type as [stdout]: either `List<int>` or `String`.
+  ///
+  /// See also:
+  /// * [FileExtension.readStringLikeShell], for reading a file with the semantics of `$(cat …)`.
+  /// * the Bash manual on command substitution: <https://www.gnu.org/software/bash/manual/bash.html#Command-Substitution>.
+  dynamic get shellOutput {
```

Comment:
Avoid `dynamic` https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic

```suggestion
  Object get shellOutput {
```